arithmetization/src/test/java/net/consensys/linea/sequencer/txselection/selectors/ProfitableTransactionSelectorTest.java

```diff
@@ -0,0 +1,405 @@
+        .isTrue();
+    assertThat(
+            transactionSelector.isTxUnprofitable(
+                mockEvaluationContext2.getPendingTransaction().getTransaction().getHash()))
+        .isTrue();
+
+    // simulate another block
+    transactionSelector = newSelectorForNewBlock();
+    // we need to decrease the min gas price in order to allow a retry
+    var minGasPriceBlock2 = Wei.of(1_000_000_000).subtract(1);
+
+    // we should remember of the unprofitable txs for the new block
+    assertThat(
+            transactionSelector.isTxUnprofitable(
+                mockEvaluationContext1.getPendingTransaction().getTransaction().getHash()))
+        .isTrue();
+    assertThat(
+            transactionSelector.isTxUnprofitable(
+                mockEvaluationContext2.getPendingTransaction().getTransaction().getHash()))
+        .isTrue();
+
+    // second try of the first tx
+    verifyTransactionSelection(
+        transactionSelector,
+        mockEvaluationContext1.setMinGasPrice(minGasPriceBlock2),
+        mockTransactionProcessingResult1,
+        SELECTED,
+        TX_UNPROFITABLE);
+
+    // second try of the second tx is not retried since we reached the retry limit
+    verifyTransactionSelection(
+        transactionSelector,
+        mockEvaluationContext2.setMinGasPrice(minGasPriceBlock2),
+        mockTransactionProcessingResult2,
+        TX_UNPROFITABLE_RETRY_LIMIT,
+        null);
+  }
+
+  @Test
+  public void shouldNotRetryUnprofitableTxWhenMinGasPriceNotDecreased() {
+    var minGasPriceBlock1 = Wei.of(1_000_000_000);
+    var mockTransactionProcessingResult1 = mockTransactionProcessingResult(21000);
+    var mockEvaluationContext1 =
+        mockEvaluationContext(false, 1000, Wei.of(1_100_000_000), minGasPriceBlock1, 210000);
+    // first try of first tx
+    verifyTransactionSelection(
+        transactionSelector,
+        mockEvaluationContext1,
+        mockTransactionProcessingResult1,
+        SELECTED,
+        TX_UNPROFITABLE);
+
+    assertThat(
+            transactionSelector.isTxUnprofitable(
+                mockEvaluationContext1.getPendingTransaction().getTransaction().getHash()))
+        .isTrue();
+
+    // simulate another block
+    transactionSelector = newSelectorForNewBlock();
+    // we keep the min gas price the same to avoid retry
+    var minGasPriceBlock2 = minGasPriceBlock1;
+
+    // we should remember of the unprofitable txs for the new block
+    assertThat(
+            transactionSelector.isTxUnprofitable(
+                mockEvaluationContext1.getPendingTransaction().getTransaction().getHash()))
+        .isTrue();
+
+    // second try of the first tx
+    verifyTransactionSelection(
+        transactionSelector,
+        mockEvaluationContext1.setMinGasPrice(minGasPriceBlock2),
+        mockTransactionProcessingResult1,
+        TX_UNPROFITABLE_MIN_GAS_PRICE_NOT_DECREASED,
+        null);
+  }
+
```

Comment:
Can we have a test case, that would see in pre-processing result as `TX_UNPROFITABLE_UPFRONT`? This new result is quite critical to this next feature but I don't see any of those in the tests.

Reply:
Yes will do

Reply:
test added